### PR TITLE
Adding support for customer-chosen language model in Azure AI Studio

### PR DESCRIPTION
**Pull request by amitkalay**
## Purpose

* This PR demonstrates how customers can use any model of their choice in Azure AI Studio to do various inference tasks. The code here is very similar to the code provided in the [AOAI Custom Skills folder](https://github.com/Azure-Samples/azure-search-power-skills/tree/main/AOAICustomSkills).

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

* Test the code

```
The code here can be tested by running the api-test.http requests in the provided file. 
```

## What to Check
Verify that the following are valid
* I ran an example to request to show that the image-captioning feature works. The following output is what you should get when you run the request with the provided default payload:

```
{
  "values": [
    {
      "warnings": null,
      "errors": [],
      "recordId": "5",
      "data": {
        "generative-caption": " The image shows a red Tesla Model S electric car driving on a highway. To make this car, the following parts are required:\n\n1. Body: The car's body is made of aluminum, which is lightweight and durable.\n2. Engine: The Tesla Model S uses an electric motor as its primary source of power.\n3. Battery: The car has a lithium-ion battery pack that stores the energy needed to power the electric motor.\n4. Chassis: The car's chassis is made of steel and provides the structural support for the car.\n5. Wheels: The Tesla Model S comes with alloy wheels, which are designed to provide good traction and stability on the road.\n6. Interior: The car's interior is made of high-quality materials and features advanced technology, such as a touchscreen display and a digital dashboard.\n7. Suspension: The car's suspension system is designed to provide a smooth and comfortable ride, even on rough roads.\n8. Tires: The Tesla Model S is equipped with low rolling resistance tires, which help improve the car's range and efficiency.\n9. Brake system: The car's braking system is designed to provide strong stopping power and help maintain the car's stability during deceleration."
      }
    }
  ]
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ This project provides the following custom skills:
...
 | --- | ----------- | -------| ----------- | ----------- | ----------- |
-| [Azure Open AI Custom Skills](AOAICustomSkills/README.md) | This demonstrates custom skills to do things in AOAI deployments like text summarization and entity recognition via an API call to model hosted in Azure AI Studio | Text | ![python](https://img.shields.io/badge/language-python-orange) | ![functions](https://img.shields.io/badge/deploy-Functions-blue) | ARM Template |
-| [Azure AI Studio Entity Recognition](EntityRecognitionCustomSkill/README.md) | summarizes provided text via an API call to model hosted in Azure AI Studio | Text | ![python](https://img.shields.io/badge/language-python-orange) | ![functions](https://img.shields.io/badge/deploy-Functions-blue) | ARM Template |
+| [Azure AI Studio Open AI Custom Skills](AOAICustomSkills/README.md) | This demonstrates custom skills to do things in AOAI deployments like text summarization and entity recognition via an API call to an Open AI language model hosted in Azure AI Studio | Text | ![python](https://img.shields.io/badge/language-python-orange) | ![functions](https://img.shields.io/badge/deploy-Functions-blue) | ARM Template |
```
Comment (steeler18) [inline README.md:18]: Remove "this" to be consistent with other rows.

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "recommendations": [
```
Comment (steeler18) [inline .vscode/extensions.json]: How about the rest client extension?

File: NonAOAICustomSkills/custom_prompts.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "summarize-default-system-prompt": "You are a useful AI assistant who is an expert at succinctly summarizing long form text into a simple summary. Summarize the text given to you in about 200 words or less.",
+    "entity-recognition-default-system-prompt": "You are a useful AI assistant. I need you to help me recognize entities in this piece of text. From the text given to you, identity all people names, addresses, email addresses, engineering job titles and present them as individual lists in a JSON object.",
+    "image-captioning-machine-info-default-prompt": "You are a useful AI assistant who is an expert in machines. You will identify all the parts for the machines in the image sent to you and will formulate the response as a list in JSON.",
```
Comment (steeler18) [inline NonAOAICustomSkills/custom_prompts.json:4]: make it consistent with other prompts?

**Review by steeler18 — APPROVED** [review]

File: NonAOAICustomSkills/custom_prompts.json (hunk 3 above)
Comment (amitkalay) [reply NonAOAICustomSkills/custom_prompts.json:4]: the prompts are the exact same across AOAI and Non-AOAI!

File: .vscode/extensions.json (hunk 2 above)
Comment (amitkalay) [reply .vscode/extensions.json]: I wasn't supposed to add this in the PR, so I deleted it.

File: README.md (hunk 1 above)
Comment (amitkalay) [reply README.md]: done

**Pull request merged**